frontend/src/HOC/withDataFetching.tsx

```diff
@@ -2,6 +2,7 @@ import React, { useEffect, useState } from 'react';
 import Style from './withDataFetching.style';
 
 const withDataFetching = <Props extends any>(
+  dispatchFunctionName: string,
```

Comment:
We do not need that, it's ok to only pass a function named fetchData to your hoc. Having this behaviour just add complexity.
However, if you really want to give that functionnality to your hoc, you could set
`dispatchFunctionName: string = 'dispatchData',` by default. In order to let your user using your hoc like before (without specifying a name)